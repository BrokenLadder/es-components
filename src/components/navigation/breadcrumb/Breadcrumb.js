import React from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';
import tinycolor from 'tinycolor2';
import genId from '../../util/generateAlphaName';
import defaultTheme from '../../theme/defaultTheme';

const OrderedList = styled.ol`
  background-color: ${props => props.theme.colors.grayLightest};
  border-radius: 2px;
  line-height: 1.4;
  list-style: none;
  margin-bottom: 25px;
  margin-top: 0;
  padding: 8px 15px;

  > li > a {
    color: ${props => props.theme.colors.primary};
    text-decoration: none;
  }

  > li > a:hover {
    color: ${props =>
      tinycolor(props.theme.colors.primary)
        .darken(15)
        .toRgbString()};
    text-decoration: underline;
  }

  > li + li {
    padding-left: 1.4em;

    &::before {
      border-color: ${props => props.theme.colors.gray};
      border-style: solid;
      border-width: 0 2px 2px 0;
      content: '';
      display: block;
      height: 13px;
      left: 0px;
      position: absolute;
      top: 20%;
      transform: rotate(-45deg);
      width: 13px;
    }
  }
`;

const Crumb = styled.li`
  color: ${props => props.theme.colors.accent};
  display: inline-block;
  position: relative;
`;

function Breadcrumb({ children, ...props }) {
  return (
    <ThemeProvider theme={props.theme}>
      <OrderedList {...props}>
        {React.Children.map(children, child => (
          <Crumb key={genId()}>{child}</Crumb>
        ))}
      </OrderedList>
    </ThemeProvider>
  );
}

Breadcrumb.propTypes = {
  children: PropTypes.node.isRequired,
  /**
   * Theme object used by the ThemeProvider,
   * automatically passed by any parent component using a ThemeProvider
   */
  theme: PropTypes.object
};

Breadcrumb.defaultProps = {
  theme: defaultTheme
};

export default Breadcrumb;
