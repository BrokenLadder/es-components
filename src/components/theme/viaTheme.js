const white = '#fff';
const gray1 = '#f4f4f4';
const gray2 = '#ecedee';
const gray3 = '#e4e4e4';
const gray4 = '#d8d8d8';
const gray5 = '#c5c5c5';
const gray6 = '#979797';
// const gray7 = '#707070';
const gray8 = '#585858';
const gray9 = '#444';
const black = '#000';

const grayLightest = gray1;
const grayLighter = gray2;
const grayLight = gray3;
const gray = gray5;
const grayDark = gray6;
const grayDarker = gray8;
const grayDarkest = gray9;

const dflt = '#d8d8d8';
const dfltHover = '#bfbfbf';
const primary = '#007fa7';
const primaryHover = '#005874';
const info = '#069';
const infoHover = '#004466';
const success = '#298544';
const successHover = '#1d5e30';
const warning = '#b35f00';
const warningHover = '#804400';
const danger = '#c00';
const dangerHover = '#990000';
const advisor = warning;

const boxShadowLight = 'rgba(0, 0, 0, 0.075)';
const boxShadowDark = 'rgba(0, 0, 0, 0.5)';

const inputBorder = gray4;
const inputFocus = '#83bffc';
const inputBoxShadow = 'rgba(102, 175, 233, 0.6)';

// Datepicker colors
const navArrow = gray;
const navArrowHover = grayDark;
const selected = '#216ba5';
const hover = '#1d5d90';
const keyboard = '#2a87d0';
const inRange = 'rgba(33, 107, 165, 0.5)';
const highlight = '#3dcc4a';
const highlightHover = '#32be3f';

const theme = {
  colors: {
    black,
    grayDarkest,
    grayDarker,
    grayDark,
    gray,
    grayLight,
    grayLighter,
    grayLightest,
    white,
    dflt,
    dfltHover,
    primary,
    primaryHover,
    success,
    successHover,
    info,
    infoHover,
    warning,
    warningHover,
    danger,
    dangerHover,
    advisor,
    boxShadowLight,
    boxShadowDark,
    inputFocus
  },
  datepickerColors: {
    navArrow,
    navArrowHover,
    selected,
    hover,
    keyboard,
    inRange,
    highlight,
    highlightHover
  },
  buttonStyles: {
    buttonsNormal: {
      primary: {
        bgColor: primary,
        textColor: white,
        hoverBgColor: primaryHover,
        hoverTextColor: white,
        activeBgColor: primaryHover,
        activeTextColor: white,
        boxShadowColor: primaryHover
      },
      secondary: {
        bgColor: dflt,
        textColor: white,
        hoverBgColor: dfltHover,
        hoverTextColor: white,
        activeBgColor: dfltHover,
        activeTextColor: white,
        boxShadowColor: dfltHover
      },
      success: {
        bgColor: success,
        textColor: white,
        hoverBgColor: successHover,
        hoverTextColor: white,
        activeBgColor: successHover,
        activeTextColor: white,
        boxShadowColor: successHover
      },
      info: {
        bgColor: info,
        textColor: white,
        hoverBgColor: infoHover,
        hoverTextColor: white,
        activeBgColor: infoHover,
        activeTextColor: white,
        boxShadowColor: infoHover
      },
      warning: {
        bgColor: warning,
        textColor: white,
        hoverBgColor: warningHover,
        hoverTextColor: white,
        activeBgColor: warningHover,
        activeTextColor: white,
        boxShadowColor: warningHover
      },
      danger: {
        bgColor: danger,
        textColor: white,
        hoverBgColor: dangerHover,
        hoverTextColor: white,
        activeBgColor: dangerHover,
        activeTextColor: white,
        boxShadowColor: dangerHover
      }
    },
    buttonsAlt: {
      primary: {
        bgColor: white,
        textColor: primary,
        hoverBgColor: primary,
        hoverTextColor: white,
        activeBgColor: primaryHover,
        activeTextColor: white,
        borderColor: primary
      },
      gray: {
        bgColor: white,
        textColor: '#63666a',
        hoverBgColor: '#63666a',
        hoverTextColor: white,
        activeBgColor: '#4a4d50',
        activeTextColor: white,
        borderColor: '#63666a'
      },
      magenta: {
        bgColor: white,
        textColor: '#c110a0',
        hoverBgColor: '#c110a0',
        hoverTextColor: white,
        activeBgColor: '#920C79',
        activeTextColor: white,
        borderColor: '#c110a0'
      },
      violet: {
        bgColor: white,
        textColor: '#702082',
        hoverBgColor: '#702082',
        hoverTextColor: white,
        activeBgColor: '#4d1659',
        activeTextColor: white,
        borderColor: '#702082'
      }
    }
  },
  // buttonSizes should always have default, lg, sm, xs
  buttonSizes: {
    lg: {
      borderRadius: '3px',
      fontSize: '26px',
      padding: '8px 20px 5px'
    },
    default: {
      borderRadius: '2px',
      fontSize: '18px',
      padding: '5px 15px 4px'
    },
    sm: {
      borderRadius: '2px',
      fontSize: '14px',
      padding: '4px 10px 3px'
    },
    xs: {
      borderRadius: '2px',
      fontSize: '12px',
      padding: '3px 10px 2px'
    }
  },
  validationIconName: {
    success: 'ok',
    warning: 'warning-sign',
    danger: 'remove'
  },
  validationTextColor: {
    success,
    warning: warningHover,
    danger
  },
  validationInputColor: {
    success: {
      borderColor: success,
      boxShadow: `inset 0 0 5px ${success}`,
      focusBorderColor: success,
      focusBoxShadow: `0 0 6px ${success}`
    },
    warning: {
      borderColor: warning,
      boxShadow: `inset 0 0 5px ${warning}`,
      focusBorderColor: warning,
      focusBoxShadow: `0 0 6px ${warning}`
    },
    danger: {
      borderColor: danger,
      boxShadow: `inset 0 0 5px ${danger}`,
      focusBorderColor: danger,
      focusBoxShadow: `0 0 6px ${danger}`
    },
    default: {
      borderColor: inputBorder,
      boxShadow: `inset 0 1px 1px ${boxShadowLight}`,
      focusBorderColor: inputFocus,
      focusBoxShadow: `0 0 8px ${inputBoxShadow}`
    }
  },
  sizes: {
    baseFontSize: 18,
    baseLineHeight: 24
  },
  screenSize: {
    retina: '320px',
    phone: '480px',
    tablet: '768px',
    desktop: '992px',
    widescreen: '1200px'
  }
};

export default theme;
