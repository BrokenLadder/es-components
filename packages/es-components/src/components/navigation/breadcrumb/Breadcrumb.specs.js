/* eslint-env jest */

import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import viaTheme from 'es-components-via-theme';
import Breadcrumb from './Breadcrumb';

it('renders as expected', () => {
  const { container } = render(
    <ThemeProvider theme={viaTheme}>
      <Breadcrumb>
        <a href="/test">test</a>
        <a href="/test-2">test 2</a>
        <a href="/test-3">test 3</a>
      </Breadcrumb>
    </ThemeProvider>
  );

  expect(container).toMatchSnapshot();
});
