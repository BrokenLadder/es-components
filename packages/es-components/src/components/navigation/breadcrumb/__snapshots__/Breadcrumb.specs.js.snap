// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbTestSuite renders as expected 1`] = `
<ol
  className="es-breadcrumb sc-bdVaJa doUOMs"
>
  <li
    className="sc-bwzfXH jyqTmE"
  >
    <a
      href="/test"
    >
      test
    </a>
  </li>
  <li
    className="sc-bwzfXH jyqTmE"
  >
    <a
      href="/test2"
    >
      test2
    </a>
  </li>
</ol>
`;
